test/e2e/es/volume_test.go

```diff
@@ -21,3 +31,115 @@ func TestVolumeEmptyDir(t *testing.T) {
 	test.Sequence(nil, test.EmptySteps, b).
 		RunSequential(t)
 }
+
+// TestVolumeExpansion resizes an existing pvc and ensures Elasticsearch
+// correctly reports the resized filesystem.
+func TestVolumeExpansion(t *testing.T) {
+	// Is there a storage class we can use that supports volume expansion?
+	// Otherwise skip this test.
+	storageClass, err := getResizeableStorageClass(test.NewK8sClientOrFatal().Client)
+	require.NoError(t, err)
+
+	if storageClass == "" {
+		t.Skip("No storage class allowing volume expansion found. Skipping the test.")
+	}
+	t.Log(fmt.Sprintf("Using storage class %s to test volume expansion", storageClass))
+	b := elasticsearch.NewBuilder("test-volume-expansion").
+		WithESMasterNodes(1, elasticsearch.DefaultResources).
+		WithESDataNodes(2, elasticsearch.DefaultResources)
+	patchStorageClasses(&b.Elasticsearch, storageClass)
+
+	masterSset := esv1.StatefulSet(b.Elasticsearch.Name, b.Elasticsearch.Spec.NodeSets[0].Name)
+	dataSset := esv1.StatefulSet(b.Elasticsearch.Name, b.Elasticsearch.Spec.NodeSets[1].Name)
+	pvcNames := []string{
+		fmt.Sprintf("elasticsearch-data-%s-0", masterSset),
+		fmt.Sprintf("elasticsearch-data-%s-0", dataSset),
+		fmt.Sprintf("elasticsearch-data-%s-1", dataSset),
+	}
+
+	// resize the volume to 1Gi after the cluster is up
+	initialStorageSize := b.Elasticsearch.Spec.NodeSets[0].VolumeClaimTemplates[0].Spec.Resources.Requests.Storage()
+	resizedStorage := initialStorageSize.DeepCopy()
+	resizedStorage.Add(resource.MustParse("1Gi"))
+
+	test.Sequence(nil, func(k *test.K8sClient) test.StepList {
+		return test.StepList{
+			{
+				Name: "Update the Elasticsearch spec with increased storage requests",
+				Test: test.Eventually(func() error {
```

Comment:
Are we using `test.Eventually` to retry on update conflict ?

Reply:
Yes! Also to account for any network glitch.